tephra-core/src/main/java/co/cask/tephra/distributed/ElasticPool.java

```diff
@@ -107,22 +111,23 @@ public ElasticPool(int sizeLimit) {
    *
    * @return an element
    */
-  public T obtain() throws E {
-    T element = getOrCreate();
-    while (true) {
-      if (element != null) {
-        return element;
-      }
-      synchronized (this) {
-        try {
-          this.wait();
-        } catch (InterruptedException e) {
-          LOG.warn("Wait interrupted. Don't know what to do. Ignoring.");
-          continue;
-        }
-        // got notified - try to get element
-        element = getOrCreate();
+  public T obtain() throws E, TimeoutException, InterruptedException {
```

Comment:
Example of common API for blocking call, you may refer to the contract of `Future.get()` and `Future.get(long timeout, TimeUnit unit)`.